app/controllers/PersonalDetailsController.scala

```diff
@@ -106,12 +109,14 @@ class PersonalDetailsController(applicationClient: ApplicationClient, userManage
       errorForm => Future.successful(Ok(views.html.application.generalDetails(errorForm, continuetoTheNextStep(onSuccess)))),
       gd => for {
         _ <- applicationClient.updateGeneralDetails(user.application.applicationId, user.user.userID,
-          removePostCodeWhenOutsideUK(gd), user.user.email, continuetoTheNextStep(onSuccess))
+          removePostCodeWhenOutsideUK(gd).toExchange(user.user.email, Some(continuetoTheNextStep(onSuccess))))
```

Comment:
Lets rename user to cachedData, these user.user's in old code should never have happened and we probably shouldn't make new ones :-)